### PR TITLE
Create namespaces before external secrets

### PR DESCRIPTION
**Pull request by enarha**
ExternalSecrets are creates at sync-wave "-1" and they are required for services like Tekton Results and OSP. But for the secrets to be created, the namespace should exist in advance. Here we create the namespace before the secret. It does not affect the deployment of the OSP operator, it ignores the fact the namespace it tries to create already exist.

### CODE REVIEW

**Comment by enarha** [comment]
It looks like sync-wave "-1" should be enough (no need for "-2"), because "By kind (e.g. namespaces first and then other Kubernetes resources, followed by custom resources)".

**Review by gabemontero — APPROVED** [review]
/lgtm

I at least @enarha am not able to tail the test logs live from the `tekton-ci` ns on the prd cluster from the ocp console.

Are you able to see the logs?

Or were you able to use dev-setup.sh or manual oc apply -k or kustomize invocations and confirm the order ?

thanks

**Comment by enarha** [comment]
@gabemontero I'm able to see the logs from the CI, but I think they are not very helpful in this case because we do not use those external secrets in any way in pipeline-service.
```
[ pipeline-service] -> kustomize build operator/gitops/argocd/pipeline-service/ | oc apply -f -
Warning: resource namespaces/openshift-pipelines is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by oc apply. oc apply should only be used on resources created declaratively by either oc create --save-config or oc apply. The missing annotation will be patched automatically.
namespace/openshift-pipelines configured
namespace/tekton-results created
serviceaccount/chains-secrets-admin created
```
shows that the namespaces are created indeed first. Specifically the openshift-pipelines should not have been there without this change.

**Comment by gabemontero** [comment]
> @gabemontero I'm able to see the logs from the CI, but I think they are not very helpful in this case because we do not use those external secrets in any way in pipeline-service.
> 
> ```
> [ pipeline-service] -> kustomize build operator/gitops/argocd/pipeline-service/ | oc apply -f -
> Warning: resource namespaces/openshift-pipelines is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by oc apply. oc apply should only be used on resources created declaratively by either oc create --save-config or oc apply. The missing annotation will be patched automatically.
> namespace/openshift-pipelines configured
> namespace/tekton-results created
> serviceaccount/chains-secrets-admin created
> ```
> 
> shows that the namespaces are created indeed first. Specifically the openshift-pipelines should not have been there without this change.

yeah the creation order was all I was worried about it it was not clear earlier @enarha ... thanks for confirming 

I'll defer you if you want to get approval from @Roming22 before merging

File: operator/gitops/argocd/pipeline-service/openshift-pipelines/namespace.yaml
Comment (Roming22) [inline operator/gitops/argocd/pipeline-service/openshift-pipelines/namespace.yaml:2]: Can you add a comment to record why that needs to happen?
I'm worried we might ask ourselves in a few months and not remember why.

Comment (enarha) [reply operator/gitops/argocd/pipeline-service/openshift-pipelines/namespace.yaml:3]: Added a comment pointing to the Jira issue as I found we did in other cases.

File: operator/gitops/argocd/pipeline-service/tekton-results/namespace.yaml
Comment (gabemontero) [inline operator/gitops/argocd/pipeline-service/tekton-results/namespace.yaml:9]: should we add a comment here too since a -1 sync-wave is "atypical" IIUC @enarha @Roming22 ?

Comment (enarha) [reply operator/gitops/argocd/pipeline-service/tekton-results/namespace.yaml:10]: Done.

**Comment by gabemontero** [comment]
assuming the upgrade failure is a flake, but let's try again 

/retest

**Comment by gabemontero** [comment]
clean / merging

**Pull request merged**
